staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -61,7 +68,7 @@ func (w *realTimeoutFactory) TimeoutCh() (<-chan time.Time, func() bool) {
 
 // serveWatch will serve a watch response.
 // TODO: the functionality in this method and in WatchServer.Serve is not cleanly decoupled.
-func serveWatch(watcher watch.Interface, scope *RequestScope, mediaTypeOptions negotiation.MediaTypeOptions, req *http.Request, w http.ResponseWriter, timeout time.Duration) {
+func serveWatch(watcher watch.Interface, scope *RequestScope, mediaTypeOptions negotiation.MediaTypeOptions, req *http.Request, w http.ResponseWriter, timeout time.Duration, opts *metainternalversion.ListOptions) {
```

Comment:
Rather than passing ListOptions here, I would prefer to actually pass the already computed `metricsScope` value.

Reply:
okay, ic, this would be computed by the `CleanListScope` function. I'm fine with that as well.